src/appsettings/validators.py

```diff
@@ -67,3 +69,43 @@ def __call__(self, value):
             if not isinstance(element, self.value_type):
                 params = {"key": key, "value": element, "type": self.value_type.__name__}
                 raise ValidationError(self.message, params=params)
+
+
+class FileValidator(object):
+    """Validator which checks file existence and open mode permissions."""
+
+    not_exist_message = "File %(value)s does not exist."
+    no_perms_message = "You have no permission to open %(value)s with mode %(mode)s."
+    unexpected_error_message = "Unexpected error occurred while opening the file %(value)s: %(exception)s"
+
+    def __init__(
+        self, exists=True, mode="r", not_exist_message=None, no_perms_message=None, unexpected_error_message=None
+    ):
+        """
+        Try to open a file using a given ``mode``.
+
+        Args:
+            exists (bool): Require existence of the file
+            mode (str): ``mode`` argument for ``open`` builtin function.
+            not_exist_message (str): Error message (containing ``value``)
+            no_perms_message (str): Error message (containing ``value`` and ``mode``)
+            unexpected_error_message (str): Error message (containing ``value`` and ``exception``)
+        """
+        self.exists = exists
```

Comment:
If `exists` is False, then the validator doesn't make any sense. If I don't want to check the existence of a file, then I shouldn't use the validator at all.

Reply:
I disagree. You may want to open a non-existing file with write mode in a folder where you have no write permissions. The open then fails with `PermissionError`.

`open('no-w-permissions-dir/example.txt', mode='r')`

Reply:
It doesn't seem that `exists` would have any effect on that case. It will fail with `no_perms_message` regardless of `exists` value flag.

Reply:
> If I don't want to check the existence of a file, then I shouldn't use the validator at all.

But you want to use the validator despite you don't care about the existence.

Reply:
Exactly. I don't see the case, where the `exists` would make a difference, if I can just not use the validator at all.